Sources/Entities/UserImpl.swift

```diff
@@ -12,9 +12,31 @@ import Foundation
 import PubNubChat
 import PubNubSDK
 
+/// A concrete implementation of the ``User`` protocol.
+///
+/// This class provides a ready-to-use solution for most use cases requiring
+/// the features defined by the ``User`` protocol, offering default behavior for
+/// associated types and default parameter values where applicable.
+///
+/// It inherits all the documentation for methods defined in the ``User`` protocol.
+/// Refer to the ``User`` protocol for detailed information on how individual methods work.
 public final class UserImpl {
   let user: PubNubChat.User
 
+  /// Creates a new ``UserImpl`` object
+  ///
+  /// - Parameters:
+  ///   - chat: Reference to the main chat object
+  ///   - id: Unique identifier for user
+  ///   - name: Display name or username of the user (must not be empty or consist only of whitespace characters)
+  ///   - externalId: Identifier for the user from an external system, such as a third-party authentication provider or a user directory
+  ///   - profileUrl: URL to the user's profile or avatar image
+  ///   - email: User's email address
+  ///   - custom: Any custom properties or metadata associated with the user in the form of a map of key-value pairs
+  ///   - status: Current status of the user, like online, offline, or away
+  ///   - type: Type of the user, like admin, member, guest
+  ///   - updated: The last updated timestamp for the object
+  ///   - lastActiveTimestamp: Timestamp for the last time the user information was updated or modified
   public convenience init(
```

Comment:
out of curiosity, your Impl classes are public with public constructors - would the developer building an app ever need to use them? normally they should only get Channels, Users and other types of objects by calling methods on Chat, Channel, etc.

Reply:
I thought these constructors are also public in Kotlin. Anyway, you're right. Users should get entities by calling methods on Chat, Channel, etc. I'll fix it

Reply:
nope, anything that's in pubnub-chat-impl is completely invisible to the developer, they ONLY see interfaces from pubnub-chat-api and some functions (like `Chat.Companion.init()`) from pubnub-chat